fhir-server/src/main/java/com/ibm/fhir/server/resources/FHIRResource.java

```diff
@@ -610,9 +653,50 @@ public Response read(@PathParam("type") String type, @PathParam("id") String id)
         try {
             checkInitComplete();
             MultivaluedMap<String, String> queryParameters = uriInfo.getQueryParameters();
+            long modifiedSince = parseIfModifiedSince();
+            
+            String ifNoneMatch = httpHeaders.getHeaderString(HEADERNAME_IF_NONE_MATCH);
+            
             Resource resource = doRead(type, id, true, false, null, null, queryParameters);
-            ResponseBuilder response = Response.ok().entity(resource);
-            response = addHeaders(response, resource);
+            int version2Match = -1;
+            // Support ETag value with or without " (and W/)
+            // e.g:  1, "1", W/1, W/"1" (the first format is used by TouchStone)
+            if (ifNoneMatch != null && !ifNoneMatch.replaceAll("\"", "").replaceAll("W/", "").trim().isEmpty()) {
```

Comment:
I think this part should be done one time, not twice as on 664/666

Reply:
I thought about this also, and because this requires another if block, so, I didn't. make sense to parse only once, will change.

Reply:
done